File: LeetCode/226翻转二叉树.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */


struct TreeNode* mirrorTree(struct TreeNode* root){
    if (root == NULL) return NULL;
    struct TreeNode *l = mirrorTree(root->left);
    struct TreeNode *r = mirrorTree(root->right);
    root->left = r;
    root->right = l;
    return root;
}

/**
 * C++做法
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        stack<TreeNode*> sta;
        if (root == nullptr) return root;
        sta.push(root);
        while (!sta.empty()) {
            TreeNode* node = sta.top();
            sta.pop();
            swap(node->left, node->right);
            if (node->right) sta.push(node->right);
            if (node->left) sta.push(node->left);
        }
        return root;
    }
};

File: LeetCode/226翻转二叉树.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
	var que []*TreeNode
	if root == nil {
		return root
	}
	que = append(que, root)
	for len(que) > 0 {
		size := len(que)
		for i := 0; i < size; i++ {
			node := que[0]
			que = que[1:]
			node.Left, node.Right = node.Right, node.Left
			if node.Left != nil {
				que = append(que, node.Left)
			}
			if node.Right != nil {
				que = append(que, node.Right)
			}
		}
	}
	return root
}